pkg/kubernetes/client_test.go

```diff
@@ -142,6 +142,7 @@ var _ = Describe("Test Namespace KubeController Methods", func() {
 			serviceChannel := events.GetPubSubInstance().Subscribe(announcements.ServiceAdded,
 				announcements.ServiceDeleted,
 				announcements.ServiceUpdated)
+			defer events.GetPubSubInstance().Close(serviceChannel)
```

Comment:
What's the reason for this? Is it to reduce the number of zombie goroutines?
Ideally, the reader never closes the chan, its the writer that initiates the close.

Reply:
discussed offline, reader still needs a way to tell pubsub to stop publishing on that channel and release it from it's tentacles as a close on reader side doesn't seem to trigger a release of it on the pubsub backend side.